test/util/testnode/utils.go

```diff
@@ -22,6 +23,12 @@ import (
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 )
 
+var (
+	// portCounter is a global atomic counter for deterministic port allocation
+	// Starting from 20000 to avoid conflicts with common ports
+	portCounter int64 = 20000
```

Comment:
just make this an atomic as that's more straightforward than having something else use the atomic package to edit this value